mobile/library/common/network/proxy_settings.h

```diff
@@ -5,17 +5,23 @@
 namespace Envoy {
 namespace Network {
 
-struct ProxySettings;
+class ProxySettings;
 using ProxySettingsConstSharedPtr = std::shared_ptr<const ProxySettings>;
 
 /**
  * Proxy settings coming from platform specific APIs, i.e. ConnectivityManager in
  * the case of Android platform.
  *
+ * ProxySettings represents already-resolved proxy settings. For example, if the proxy
+ * settings are obtained from a PAC URL, then this object represents the resolved proxy
+ * settings after reading and parsing the PAC URL.
+ *
+ * TODO(abeyad): rename this class to ProxySetting (the plural in the name is confusing).
  */
-struct ProxySettings {
+class ProxySettings {
```

Comment:
The implementation should go to `.cc` IMO.

Reply:
I agree, and at first, I wanted to make this change separately, but I actually reworked some things and moved more types to proxy_settings.h, so I separated the implementation out into a separate .cc file

Reply:
SGTM. Thanks.